enhancements/profile-versioning.md

```diff
@@ -0,0 +1,245 @@
+using the `metadata.version` attribute of the profile file. During the content
+build process, this version is populated in the data stream.
+
+The Compliance Operator profile parser will look for this attribute in the data
+stream while it processes the content at installation time. If a profile in the
+data stream contains `metadata.version`, the profile parser will persist the
+`Profile` CRD with a version attribute set.
+
+Because the profile version is ultimately coming from the content, we need to
+have a process or automation in place to ensure the profile version is accurate
+across releases.
+
+### User Stories
+
+1. As a cluster administrator, I want to be able to see the version of a
+   particular profile supported by the Compliance Operator.
+2. As a cluster administrator, I want to be able to see if the content used
+   across Compliance Operator releases changed based on the `version`.
+
+### API Extensions
+
+This enhancement focuses solely on adding a `version` property to the `Profile`
+CRD. No other resources will be modified as a result of this implementation.
+
+The implementation should be backwards compatible so that if a profile does not
+have a version the `Profile` will have an empty value for `version`.
+
+### Version Evaluation
+
+Profile version semantics should follow [semantic versioning v2.0.0](https://semver.org/spec/v2.0.0.html),
+which is the most current supported version of SemVer at the time of this
+writing. A summary of the differences between version 1.0.0 and 2.0.0 are
+available in a [GitHub issue](https://github.com/semver/semver/issues/231).
+
+This would allow us to use the following strings to represent profile versions:
+
+* `1.2.0` being the most basic form of versions we see in compliance benchmarks
+* `1.2.0-1` includes flexibility to denote differences across the same profile version
+* `1.2.0-1+e076b1d7ac` include the content of the profile as a SHA
+
+Initially, we may choose to only implement simple versions (e.g., 1.2.0, or
+4.0.0). Depending on user feedback, we have the option to include support for
+more complex version strings that use pre-release or build metadata within the
+semantic version specification to denote changes between operator releases.
+
+Note that if we do decide to support pre-release or build metadata as part of
+the version string, we should ensure version string evaluation considers
+`1.2.0` older than `1.2.0-1` or `1.2.0-2+e076b1d7ac`.
+
+### Implementation Details/Notes/Constraints [optional]
+
+What are the caveats to the implementation? What are some important details that
+didn't come across above. Go in to as much detail as necessary here. This might
+be a good place to talk about core concepts and how they relate.
+
+### Risks and Mitigations
+
+What are the risks of this proposal and how do we mitigate. Think broadly. For
+example, consider both security and how this will impact the larger OKD
+ecosystem.
+
+How will security be reviewed and by whom?
```

Comment:
What do you mean by security here?

Reply:
This is really just boilerplate questions for the enhancement template.

Although - from a security perspective, users should ensure the profile version they're using is what they intend to run. For example at the time of this writing, running CIS OpenShift 1.6.0 wouldn't be possible, because that profile version doesn't exist.